main/src/test/java/eu/openg/guessnumberapi/main/GetGamesIntegrationTest.java

```diff
@@ -0,0 +1,53 @@
+package eu.openg.guessnumberapi.main;
+
+import eu.openg.guessnumberapi.domain.Game;
+import eu.openg.guessnumberapi.gateway.api.GameRepository;
+import eu.openg.guessnumberapi.gateway.fake.FakeGameIdProvider;
+import eu.openg.guessnumberapi.gateway.implementation.InMemoryGameRepo;
+import eu.openg.guessnumberapi.usecase.api.BoundaryGame;
+import eu.openg.guessnumberapi.usecase.api.GetGamesUseCase;
+import eu.openg.guessnumberapi.usecase.implementation.GetGamesInteractor;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class GetGamesIntegrationTest {
+    private GetGamesUseCase getGamesUseCase;
+    private GameRepository gameRepository;
+
+    @Test
+    void getGamesWhenDbIsEmpty() {
+        gameRepository = new InMemoryGameRepo(new FakeGameIdProvider());
+        getGamesUseCase = new GetGamesInteractor(gameRepository);
+
+        List<BoundaryGame> boundaryGames = getGamesUseCase.fetchGames();
+        assertEquals(Collections.emptyList(), boundaryGames);
+    }
+
+    @Test
+    void getGamesWhenDbIsNotEmpty() {
+        Map<Integer, Game> storage = new HashMap<>();
```

Comment:
storage į fieldą ir visas inicializacijas į setup. šitam teste galėsi į storage prisidėti ko reikia, o kitas naudos as it is

Reply:
Neišeis, nes dabar Marčio paliepimu pakūriau FakeRepo, kuris visada kažką grąžina ir naudoju šitam metode, o kitam seną realRepo

Reply:
td neverta turėti uc ir repository fielduose, reiktų juos į local var persikelt